src/style-cache.js

```diff
@@ -0,0 +1,50 @@
+/**
+@license
+Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+'use strict';
+
+export default class StyleCache {
+  constructor(max = 100) {
+    // map element name -> [{properties, stylesheet, scopeSelector}]
+    this.cache = {};
+    this.max = max;
```

Comment:
Let's name this more explicitly since it's a maximum per type, perhaps `typeMax` ?